File: client/src/main/java/pro/panopticon/client/awscloudwatch/HasCloudwatchConfig.java
package pro.panopticon.client.awscloudwatch;

import com.amazonaws.auth.AWSCredentials;

public interface HasCloudwatchConfig extends AWSCredentials {

    String getAWSAccessKeyId();

    String getAWSSecretKey();

    String getRegion();

    boolean auditeventStatisticsEnabled();

    boolean sensorStatisticsEnabled();

    String getAppName();

    String getEnvironment();

}

File: client/src/main/java/pro/panopticon/client/awscloudwatch/HasCloudwatchConfig.kt
package pro.panopticon.client.awscloudwatch

import com.amazonaws.auth.AWSCredentials

interface HasCloudwatchConfig : AWSCredentials {
    override fun getAWSAccessKeyId(): String
    override fun getAWSSecretKey(): String
    val region: String?
    fun auditeventStatisticsEnabled(): Boolean
    fun sensorStatisticsEnabled(): Boolean
    val appName: String?
    val environment: String?
}
